subprojects/integ-test/src/crossVersionTest/groovy/org/gradle/integtests/PropertyUpgradesBinaryCompatibilityCrossVersionSpec.groovy

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.integtests
+
+import groovy.test.NotYetImplemented
+import org.gradle.api.plugins.quality.Checkstyle
+import org.gradle.integtests.fixtures.TargetVersions
+
+@TargetVersions("8.0.2")
+class PropertyUpgradesBinaryCompatibilityCrossVersionSpec extends AbstractPropertyUpgradesBinaryCompatibilityCrossVersionSpec {
+
+    @Override
+    protected List<Class<?>> importClasses() {
+        return [Checkstyle]
+    }
+
+    @NotYetImplemented
+    def "can use upgraded Checkstyle in a Groovy plugin compiled with a previous Gradle version"() {
+        given:
+        prepareGroovyPluginTest """
+            project.tasks.register("myCheckstyle", Checkstyle) {
+                maxErrors = 1
+                // int currentMaxErrors = maxErrors doesn't work yet
+                int currentMaxErrors = it.maxErrors
+                assert currentMaxErrors == 1
+            }
+        """
+
+        expect:
+        version previous withTasks 'assemble' inDirectory(file("producer")) run()
+        version current withTasks 'tasks' withStacktraceEnabled() requireDaemon() requireIsolatedDaemons() run()
```

Comment:
🤔 I wonder how we could reduce the redundancy of having to add these two lines in every test...